src/lib/cli.rs

```diff
@@ -33,8 +35,10 @@ fn run(cli_args: CliArgs, global_config: &GlobalConfig) {
         color: !cli_args.disable_color,
     });
 
-    info!("{} {}", &cli_args.command, &VERSION);
-    debug!("Written By {}", &AUTHOR);
+    if env::var("CARGO_MAKE_INTERNAL_RECURSION").unwrap_or_else(|_| "0".to_string()) != "0" {
```

Comment:
can we move all the 'internal recursion' logic to a separate module.
i see duplicate code in many areas of this PR and feels like this should be a really small standalone module with
get, is_first and so on...
and since i saw the logger is using this as well, this new module should not have any inner dependency from other cargo-make modules. 
what do you think?

also instead of that complex line, you can do in the module envmnt::get_or("CARGO_MAKE_INTERNAL_RECURSION", "0".to_string())